walk-in-interview/src/main/java/com/google/job/data/JobPayment.java

```diff
@@ -0,0 +1,99 @@
+package com.google.job.data;
+
+/** Class that represents the payment details of a job. */
+public final class JobPayment {
+    private final int min;
+    private final int max;
+    private final PaymentFrequency paymentFrequency;
+
+    private int hashCode;
+
+    private void validateParameter(float min, float max) throws IllegalArgumentException {
+        if (min < 0) {
+            throw new IllegalArgumentException("\"min\" should not be negative");
+        }
+
+        if (max < min) {
+            throw new IllegalArgumentException("\"max\" should not be less than \"min\"");
+        }
+    }
+
+    // For serialization
+    public JobPayment() {
+        this.min = 0;
+        this.max = 0;
+        this.paymentFrequency = PaymentFrequency.HOURLY;
+    }
+
+    public JobPayment(int min, int max, PaymentFrequency paymentFrequency) {
+        validateParameter(min, max);
+
+        this.min = min;
+        this.max = max;
+        this.paymentFrequency = paymentFrequency;
+    }
+
+    /**
+     * Gets the lower limit of the payment.
```

Comment:
same comment as the other classes, you don't need `@return`, just do this:
```
/** Returns the lower limit of the payment, never negative. */
```

same comment for other files and other methods.

Reply:
Fixed